cmd/fleet/serve_test.go

```diff
@@ -23,6 +23,23 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// safeStore is a wrapper around mock.Store to allow for concurrent calling to
+// AppConfig, in the past we have seen this test fail with a data race warning.
+//
+// TODO: if we see other tests failing for similar reasons, we should build a
+// more robust pattern instead of doing this everywhere
+type safeStore struct {
+	mock.Store
+	mu sync.Mutex
+}
+
+func (s *safeStore) AppConfig(ctx context.Context) (*fleet.AppConfig, error) {
+	s.mu.Lock()
+	s.AppConfigFuncInvoked = true
```

Comment:
Do we ever assert that it is true/false somewhere in the tests? If so, that check would need to be mutex-protected too, so a func to return the field with mutex-locking should be implemented instead of directly accessing the field (I didn't find such a check from a quick glance).

Reply:
great catch, thanks, I didn't think of that! I checked and as you mention fortunately we don't read the value.